### PR TITLE
statistics: refactor the code

### PR DESCRIPTION
**Pull request by Rustin-Liu**
remove redundant return statement.
omit comparison to bool constant.
use a single append to concatenate two slices.

### What problem does this PR solve? 
#13736 

### Check List 

Tests 

None

Code changes

None

Side effects

None

Related changes

None

Release note

None

### CODE REVIEW

File: statistics/handle/update.go [hunk 1]
```diff
@@ -283,7 +283,7 @@ func (h *Handle) DumpStatsDeltaToKV(dumpMode bool) error {
 	h.sweepList()
 	currentTime := time.Now()
 	for id, item := range h.globalMap {
-		if dumpMode == DumpDelta && !needDumpStatsDelta(h, id, item, currentTime) {
+		if !dumpMode && !needDumpStatsDelta(h, id, item, currentTime) {
```
Comment (tangenta) [inline statistics/handle/update.go]: I think we should assign a type for `dumpMode` instead of using `bool`.

Comment (Rustin-Liu) [reply statistics/handle/update.go]: I think so, maybe I will add it tonight.

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

File: statistics/handle/update.go (hunk 1 above)
Comment (Rustin-Liu) [reply statistics/handle/update.go]: added.

File: statistics/handle/update.go [hunk 2]
```diff
@@ -278,8 +280,8 @@ func (h *Handle) sweepList() {
 }
 
 // DumpStatsDeltaToKV sweeps the whole list and updates the global map, then we dumps every table that held in map to KV.
-// If the `dumpAll` is false, it will only dump that delta info that `Modify Count / Table Count` greater than a ratio.
-func (h *Handle) DumpStatsDeltaToKV(dumpMode bool) error {
+// If the dumpMode is `DumpDelta`, it will only dump that delta info that `Modify Count / Table Count` greater than a ratio.
+func (h *Handle) DumpStatsDeltaToKV(dumpMode dumpMode) error {
```
Comment (wjhuang2016) [inline statistics/handle/update.go]: Use `mode` instead of `dumpMode`, since the type is `dumpMode`.

Comment (Rustin-Liu) [reply statistics/handle/update.go]: renamed.

**Comment by Rustin-Liu** [comment]
PTAL~

**Pull request merged**